tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -174,25 +390,33 @@ def _prepare_inputs(
         self,
         scheduled_requests: ScheduledRequests,
         resource_manager: ResourceManager,
-        new_tokens: Optional[torch.Tensor] = None,
+        new_tensors_device: Optional[SampleStateTensors] = None,
+        spec_metadata: Optional[SpecMetadata] = None,
     ) -> List[bool]:
         """Prepare inputs for AD Model from scheduled requests."""
         # cache manager
+
+        new_tokens = new_tensors_device.new_tokens if new_tensors_device is not None else None
+
+        # TODO(govind): Manage guided decoding + speculative decoding together.
+
         kv_cache_manager = resource_manager.get_resource_manager(
             ResourceManagerType.KV_CACHE_MANAGER
         )
 
         # requests in order of context, generate
         context_requests = scheduled_requests.context_requests
-        gen_requests = [r for r in scheduled_requests.generation_requests if not r.draft_tokens]
+        gen_requests = scheduled_requests.generation_requests
```

Comment:
pytorch backend iterates over generation requests with this suborder

1. extend requests
2. first draft requests
3. generation requests

It may be useful later on to maintain this order